[Kiranlimtl] iP

Kiranlimtl opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Kiranlimtl committed:
Level-0

Kiranlimtl committed:
Level-1. Echo

Kiranlimtl committed:
Level-2. Add, List

Kiranlimtl committed:
Level-3.Mark As Done

Kiranlimtl committed:
Level-3.1.Mark As Done

Kiranlimtl committed:
Level-4.ToDos, Events, Deeadlines

Kiranlimtl committed:
Level-4.1.ToDos, Events, Deadlines

Kiranlimtl committed:
Level-4.2.ToDos, Events, Deadlines

Kiranlimtl committed:
Level-4.3.ToDos, Events, Deadlines

Kiranlimtl committed:
Level-5.Handle_Error

Kiranlimtl committed:
Level-6.Delete

Kiranlimtl committed:
Add javadocs and change logic for scanner

Kiranlimtl committed:
Add testing

Kiranlimtl committed:
Add test cases and expected results

Kiranlimtl committed:
Merge commit '2b59fb337ac8727bd5c0b89171d9f24d6e71a9ec'

Kiranlimtl committed:
Merge remote-tracking branch 'origin/master'

Kiranlimtl committed:
Update test

Kiranlimtl committed:
Add Storage.java
WallE.java: Add ability read and write to new file

Kiranlimtl committed:
Revert "Add Storage.java"

This reverts commit ea8c633ff80c39b41480050587e9a7e0a886600e.

Kiranlimtl committed:
Reapply "Add Storage.java"

This reverts commit a24dca7762e3fe992c73dc9bd0e5fa5415823269.

Kiranlimtl committed:
Merge branch 'branch-Level-7'

Kiranlimtl committed:
Merge branch 'master' of https://github.com/Kiranlimtl/ip

Kiranlimtl committed:
Add date and time compatibility for event and deadline

Kiranlimtl committed:
Add new class DateTimeUtility
Update Deadline and Event class to use DateTimeUtility
Update WallE class to use DateTimeUtility

Kiranlimtl committed:
Merge branch 'branch-Level-8'

Kiranlimtl committed:
Update text-ui for testing

Kiranlimtl committed:
Merge branch 'branch-Level-8'

hrishikeshsathyian commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,22 @@
+     */
+    @Override
+    public String toString() {
+        return "[T]" + super.toString() ;
```
Small nit. Perhaps we can remove the extra spacing after the toString() call.
```suggestion
        return "[T]" + super.toString();
```

hrishikeshsathyian commented on src/main/java/WallE.java:
```diff
@@ -0,0 +1,227 @@
+        if (input.startsWith("delete")) return DELETE;
+        if (input.equalsIgnoreCase("list")) return LIST;
+        return INVALID;
+    }
```
Should we be using curly braces for the single-line conditional statements?

hrishikeshsathyian replied:
Is there any reason why you chose to use chained if conditional statements, instead of a switch statement, which might be cleaner?

hrishikeshsathyian commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,22 @@
+     * Constructor for Todo class
+     * @param description
+     */
+    public ToDo(String description){
```
Should there be a space after the (
```suggestion
    public ToDo(String description) {
```

hrishikeshsathyian commented on src/main/java/WallE.java:
```diff
@@ -0,0 +1,227 @@
+        Command command = Command.fromString(userInput);
+        String commandKeyword = command == Command.MARK ? "mark" : "unmark";
+        if (userInput.length() == commandKeyword.length() || userInput.substring(commandKeyword.length()).trim().isEmpty()) {
+            throw new WallException("What do you want to " + (commandKeyword) + " huh?");
```
Should the || operator be on the next line instead?

hrishikeshsathyian reviewed (commented):
A few nits and some clarifications. Mostly looks good otherwise.

Huan-Kiat commented on src/main/java/WallE.java:
```diff
@@ -0,0 +1,227 @@
+    private static final String FILE_PATH = "./data/walle.txt";
+    private static  Storage storage;
+    private static ArrayList<Task> tasks = new ArrayList<Task>();
+    public static void main(String[] args) {
```
Perhaps we can leave a line between variables and your main method for readability?

Huan-Kiat commented on src/main/java/WallE.java:
```diff
@@ -0,0 +1,227 @@
+        scanner.close();
+    }
+
+
```
I noticed the same issue of the lack of a space between your parentheses and braces in several other places.

Huan-Kiat commented on src/main/java/WallE.java:
```diff
@@ -0,0 +1,227 @@
+    private static void handleAddedTask(ArrayList<Task> tasks, String userInput) throws WallException {
+        Command command = Command.fromString(userInput);
+        switch (command) {
+            case TODO:
```
Should there still be braces for single lined conditionals? 
Placing conditionals on a separate line will also enhance readability.

Huan-Kiat commented on src/main/java/WallE.java:
```diff
@@ -0,0 +1,227 @@
+            printHorizontalLine();
+        }
+
+    }
```
Should there be a line spacing before starting on the next method?

Huan-Kiat reviewed (commented):
There seems to be cases where there is an extensive use of conditionals, leading to a less readable code. Consider changing to switch statements or perhaps following the coding standards for the conditionals will enhance readability.

Kiranlimtl committed:
Add more OOP with Command, Parser, Ui.
Update Storage to comply with new OOP
Add different command class

Kiranlimtl committed:
Update test input

Kiranlimtl committed:
Add packages
Update code to handle package

Kiranlimtl committed:
Merge branch 'branch-A-Packages'

Kiranlimtl committed:
Revert "Merge branch 'branch-A-Packages'"

This reverts commit bc8c86f74d8263c3655619f78b8445ec724cf36d, reversing
changes made to 70ec1d880c49c52f961ec63a9fd509f80e7f84c9.

Kiranlimtl committed:
Reapply "Merge branch 'branch-A-Packages'"

This reverts commit fa8cc2aa0a313d6115cde07fe6b6894cb370de4b.

Kiranlimtl committed:
Update test

Kiranlimtl committed:
Merge branch 'branch-A-MoreOOp'

Kiranlimtl committed:
Merge branch 'add-gradle-support' of https://github.com/Kiranlimtl/ip into add-gradle-support

Kiranlimtl committed:
Merge branch 'add-gradle-support'

Kiranlimtl committed:
Update previous missing commit of exceptoin

Kiranlimtl committed:
Add 2 JUnit file

Kiranlimtl committed:
Merge branch 'branch-A-JUnit'

Kiranlimtl committed:
Update to create jar file

Kiranlimtl committed:
Merge branch 'branch-A-Jar'

Kiranlimtl committed:
Add more JavaDoc

Kiranlimtl committed:
Update code to fit coding standard

Kiranlimtl committed:
Add find command feature to find similar tasks base of string
Add more exceptions to customise exception message

Kiranlimtl committed:
Merge branch 'branch-A-JavaDoc'

Kiranlimtl committed:
Merge branch 'branch-A-CodingStandard'

Kiranlimtl committed:
Merge branch 'branch-Level-9'

Kiranlimtl committed:
Update README.md

Kiranlimtl committed:
Update README.md

Kiranlimtl committed:
Update README.md

Kiranlimtl committed:
Add checkstyle

Kiranlimtl committed:
Merge branch 'branch-A-Checkstyle'

Kiranlimtl committed:
Update README.md

Kiranlimtl committed:
Merge pull request #1 from Kiranlimtl/Use-GFMD

Update README.md

Kiranlimtl committed:
Add importthat i missed

Kiranlimtl committed:
Fix error in deleting

Kiranlimtl committed:
Add JavaFx and GUI

Kiranlimtl committed:
Merge branch 'branch-Level-10'

Kiranlimtl committed:
Add a few assertions

Kiranlimtl committed:
Fix code quality
Reposition import statements

Kiranlimtl committed:
Merge pull request #4 from Kiranlimtl/branch-A-Assertions

Add a few assertions

Kiranlimtl committed:
Fix conflicts

Kiranlimtl committed:
Fix indentation for AddCommand

Kiranlimtl committed:
Merge branch 'master' into branch-A-CodeQuality

Kiranlimtl committed:
Merge pull request #3 from Kiranlimtl/branch-A-CodeQuality

Fix code quality

Kiranlimtl committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/walle/commands/AddCommand.java

Kiranlimtl committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/Kiranlimtl/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/walle/commands/AddCommand.java

Kiranlimtl committed:
feat: Add B-Reminder feature to display tasks occurring today

- Implemented B-Reminder feature that automatically checks for tasks with
  dates matching the current day upon application startup.
- Modified `ReminderCommand` to filter tasks by `LocalDate.now()`, including
  both `Deadline` and `Event` tasks.
- Integrated reminder display into `Main` so that reminders appear in the UI
  when the app starts.
- Ensured that reminders are fetched and formatted correctly in `Ui` for
  better readability.
- Improved `Storage` and `DateTimeUtility` to handle date parsing and ensure
  reminders are loaded properly from saved tasks.
- Fixed bye input not closing the app

With this update, users will receive an automatic reminder of all tasks
happening today whenever they open the app.

Kiranlimtl committed:
Merge pull request #5 from Kiranlimtl/branch-B-Reminders

feat: Add B-Reminder feature to display tasks occurring today

Kiranlimtl committed:
Merge branch 'branch-A-CodeQuality' into branch-B-Reminders

Kiranlimtl committed:
Merge branch 'branch-B-Reminders'

Kiranlimtl committed:
Merge branch 'master' of https://github.com/Kiranlimtl/ip

Kiranlimtl committed:
Update README.md

Kiranlimtl committed:
Merge pull request #6 from Kiranlimtl/Use-GFMD-1

Update README.md

Kiranlimtl committed:
Update README.md

Added Reminders

Kiranlimtl committed:
Update README.md

Fixed gfmd bug

Kiranlimtl committed:
Updated README with comprehensive user guide, added product screenshot, and modified app colors

- Enhanced `README.md` with a detailed user guide, including installation steps, feature descriptions, and command summaries.
- Incorporated a new `product-screenshot.png` to provide users with a visual preview of the application interface.
- Modified the color scheme of the application to improve readability and aesthetics.

These changes improve user experience by offering clear documentation, a visual reference, and an updated UI design.

Kiranlimtl committed:
Merge branch 'branch-A-UserGuide'

Kiranlimtl committed:
Update the summary of command to be aligned

Kiranlimtl committed:
Update Readme.md to fix alignment of command summary

Kiranlimtl committed:
Fix Checkstyle violations in codebase

Kiranlimtl committed:
Merge branch 'branch-A-Release'

Kiranlimtl committed:
Fix bug in image loading and ensure resources are packaged

Kiranlimtl committed:
Fix image loading error due to incorrect resource path

Kiranlimtl committed:
Fix spelling error and update code example in documentation

Kiranlimtl committed:
Merge branch 'branch-A-UseGFMD'

Kiranlimtl committed:
Update README to include bug fixes

Kiranlimtl committed:
Update README to use GFMD
